File: modules/module2/src/main/java/com/aidaole/WindowInterfaceImpl.java
package com.aidaole;

import com.aidaole.injectservice.runtime.InjectManager;
import com.aidaole.lib.interfaces.DialogInterface;
import com.aidaole.lib.interfaces.WindowInterface;
import com.aidaole.injectservice.runtime.InjectService;

@InjectService(WindowInterface.class)
public class WindowInterfaceImpl implements WindowInterface {
    @Override
    public String hello() {
        DialogInterface dialog = InjectManager.get(DialogInterface.class);
        return "hello " + dialog.dialog();
    }
}

File: modules/module2/src/main/java/com/aidaole/WindowInterfaceImpl.kt
package com.aidaole

import com.aidaole.injectservice.runtime.InjectManager
import com.aidaole.injectservice.runtime.InjectService
import com.aidaole.lib.interfaces.DialogInterface
import com.aidaole.lib.interfaces.WindowInterface

@InjectService(WindowInterface::class)
class WindowInterfaceImpl : WindowInterface {
    override fun hello(): String {
        val dialog = InjectManager.get(DialogInterface::class.java)
        return "hello " + dialog.dialog()
    }
}